src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -90,16 +90,60 @@ Base.adjoint(A::LazyHcat) = Adjoint(A)
     for blA in A.blocks
         ocol = 0  # column offset
         for blB in B.blocks
-            ret[orow .+ (1:size(blA, 2)), ocol .+ (1:size(blB, 2))] .= blA' * blB
+            mul!(ret[orow .+ (1:size(blA, 2)), ocol .+ (1:size(blB, 2))], adjoint(blA), blB)
             ocol += size(blB, 2)
         end
         orow += size(blA, 2)
     end
     ret
 end
 
+# Special case of Hermitian result: can only actively compute the block upper diagonal
+@views function mul_hermi(Aadj::Adjoint{T,<:LazyHcat}, B::LazyHcat) where {T}
```

Comment:
I think you are assuming that `B::LazyHcat{T}`, i.e. both matrices are of the same element type.

Reply:
Also, I wonder if this could be handled more transparently if one had a version
`mul!(::Hermitian, ::Adjoint{T, <:LazyHcat}, ::LazyHcat)`, but maybe that's non-standard and overengineering ...